packages/asset-uploader/src/upload.ts

```diff
@@ -22,31 +21,42 @@ export const uploadAssets = async ({
 }: {
   request: Request;
   projectId: string;
-}): Promise<Asset[]> => {
-  const directory = await imageFsDirectory();
-  const formData = await unstable_parseMultipartFormData(
-    request,
-    isS3Upload
-      ? (file) =>
-          s3UploadHandler({
-            file,
+}) => {
+  try {
+    const directory = await imageFsDirectory();
+    const formData = await unstable_parseMultipartFormData(
+      request,
+      isS3Upload
+        ? (file) =>
+            s3UploadHandler({
+              file,
+              maxPartSize: MAX_UPLOAD_SIZE,
+            })
+        : unstable_createFileUploadHandler({
             maxPartSize: MAX_UPLOAD_SIZE,
+            directory,
+            file: ({ filename }) => filename,
           })
-      : unstable_createFileUploadHandler({
-          maxPartSize: MAX_UPLOAD_SIZE,
-          directory,
-          file: ({ filename }) => filename,
-        })
-  );
-  if (isS3Upload) {
-    return await uploadToS3({
-      projectId,
-      formData,
-    });
-  } else {
-    return await uploadToDisk({
-      projectId,
-      formData,
-    });
+    );
+    if (isS3Upload) {
+      return await uploadToS3({
+        projectId,
+        formData,
+      });
+    } else {
+      return await uploadToDisk({
+        projectId,
+        formData,
+      });
+    }
+  } catch (error) {
+    if ("maxBytes" in error) {
+      throw new Error(
+        `Asset cannot be bigger than ${commonUploadVars.MAX_UPLOAD_SIZE}MB`
+      );
+    }
```

Comment:
Getting a sad type error here :(

```
Object is of type 'unknown'.
```

Reply:
let me know if you need help, can check this out tomorrow

Reply:
Yeah, lets check it out tomorrow